syntax = "proto3";

service TrackerService {
  rpc RegisterPeer (PeerInfo) returns (Status);
  rpc GetPeers (FileRequest) returns (PeerList);
}

service PeerService {
  rpc RequestChunk (ChunkRequest) returns (Chunk);
}

message PeerInfo {
  string peer_id = 1;
  string address = 2;
  repeated string files = 3; // Lista de archivos compartidos
}

message FileRequest {
  string file_name = 1;
}

message PeerList {
  repeated PeerInfo peers = 1;
}

message ChunkRequest {
  string file_name = 1;
  int32 chunk_index = 2;
}

message Chunk {
  int32 chunk_index = 1;
  bytes content = 2;
}

message Status {
  string message = 1;
}
